app/core/social-handlers/GoogleClassroomHandler.js

```diff
@@ -0,0 +1,132 @@
+  constructor () {
+    application.gplusHandler.loadAPI()
+    super()
+  }
+
+  loadClassroomsFromAPI () {
+    return new Promise((resolve, reject) => {
+      gapi.client.load ('classroom', 'v1', () => {
+        gapi.client.classroom.courses.list({access_token: application.gplusHandler.token(), teacherId: me.get('gplusID'), courseStates: "ACTIVE"})
+        .then((r) => {
+          resolve(r.result.courses || [])
+          })
+        .catch ((err) => {
+          console.error("Error in fetching from Google Classroom:", err)
+          reject(err)
+          })
+        })
+      })
+  }
+
+  loadStudentsFromAPI (googleClassroomId) {
+    return new Promise((resolve, reject) => {
+      gapi.client.load ('classroom', 'v1', () => {
+        gapi.client.classroom.courses.students.list({access_token: application.gplusHandler.token(), courseId: googleClassroomId})
+        .then((r) => {
+          resolve(r.result.students || [])
+          })
+        .catch ((err) => {
+          console.error("Error in fetching from Google Classroom:", err)
+          reject(err)
+          })
+        })
+      })
+  }
+
+}
+
+module.exports = {
+  gcApiHandler: new GoogleClassroomAPIHandler(),
+
+  scopes: 'https://www.googleapis.com/auth/classroom.courses https://www.googleapis.com/auth/classroom.courses.readonly https://www.googleapis.com/auth/classroom.profile.emails https://www.googleapis.com/auth/classroom.profile.photos https://www.googleapis.com/auth/classroom.rosters https://www.googleapis.com/auth/classroom.rosters.readonly',
+
+  markAsImported: async function(gcId) {
+    try {
+      let gClass = me.get('googleClassrooms').find((c)=>c.id==gcId)
+      if (gClass) {
+        gClass.importedToCoco = true
+        await new Promise(me.save().then)
+      }
+      else {
+        return Promise.reject("Classroom not found in me.googleClassrooms")
+      }
+    }
+    catch (err) {
+      console.error("Error in marking classroom as imported:", err)
+      return Promise.reject("Error in marking classroom as imported")
+    }
+  },
+
+  importClassrooms: async function() {
+    try {
+      const importedClassrooms = await this.gcApiHandler.loadClassroomsFromAPI()
+      const importedClassroomsNames = importedClassrooms.map((c) => {
+        return { id: c.id, name: c.name }
+      })
+     
+      const classrooms = (me.get('googleClassrooms') || []).filter((c) => c.importedToCoco == true)
+      importedClassroomsNames.forEach((imported) => {
+        if (!classrooms.find((c) => c.id == imported.id)) {
+          classrooms.push(imported)
+        }
+      })
+      me.set('googleClassrooms', classrooms)
+      await new Promise(me.save().then)
+    }
+    catch (err) {
+      console.error("Error in importing classrooms", err)
+      return Promise.reject()
+    }
+  },
+  
+  // Imports students from google classroom, create their account on coco and add to the coco classroom
+  importStudentsToClassroom: async function (cocoClassroom) {
```

Comment:
For this method should we contain the try catch blocks to be more specific rather than the majority of the method.

The benefits of this would be getting more feedback on what exactly went wrong. It may also allow the opportunity to cleanup or retry after the failure.

I suppose the only method here that makes a big difference is the addMembers method at the end so this looks fine.